### ISSUE TITLE
iOS: Event type cannot be null, empty or blank (Push notifications)

### ISSUE THREAD

**Issue by tyndria** [aws-amplify/amplify-js#11097]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Authentication, GraphQL API, Push Notifications

### Amplify Categories

auth, api, notifications

### Environment information

<details>

```
# Put output below this line
  System:
    OS: macOS 12.6.2
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
    Memory: 330.56 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 17.6.0 - ~/.nvm/versions/node/v17.6.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v17.6.0/bin/yarn
    npm: 8.5.1 - ~/.nvm/versions/node/v17.6.0/bin/npm
    Watchman: 2023.02.13.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 111.0.5563.64
    Safari: 15.6.1
    Safari Technology Preview: 14.2
  npmPackages:
    @apollo/client: ^3.7.7 => 3.7.7 
    @apollo/client/cache:  undefined ()
    @apollo/client/core:  undefined ()
    @apollo/client/errors:  undefined ()
    @apollo/client/link/batch:  undefined ()
    @apollo/client/link/batch-http:  undefined ()
    @apollo/client/link/context:  undefined ()
    @apollo/client/link/core:  undefined ()
    @apollo/client/link/error:  undefined ()
    @apollo/client/link/http:  undefined ()
    @apollo/client/link/persisted-queries:  undefined ()
    @apollo/client/link/retry:  undefined ()
    @apollo/client/link/schema:  undefined ()
    @apollo/client/link/subscriptions:  undefined ()
    @apollo/client/link/utils:  undefined ()
    @apollo/client/link/ws:  undefined ()
    @apollo/client/react:  undefined ()
    @apollo/client/react/components:  undefined ()
    @apollo/client/react/context:  undefined ()
    @apollo/client/react/hoc:  undefined ()
    @apollo/client/react/hooks:  undefined ()
    @apollo/client/react/parser:  undefined ()
    @apollo/client/react/ssr:  undefined ()
    @apollo/client/testing:  undefined ()
    @apollo/client/testing/core:  undefined ()
    @apollo/client/utilities:  undefined ()
    @apollo/client/utilities/globals:  undefined ()
    @aws-amplify/pushnotification: ^5.0.20 => 5.0.20 
    @aws-sdk/client-lex-models-v2: ^3.262.0 => 3.279.0 
    @aws-sdk/client-lex-runtime-v2: ^3.262.0 => 3.279.0 (3.186.0)
    @babel/core: ^7.12.9 => 7.21.0 
    @babel/runtime: ^7.12.5 => 7.21.0 
    @notifee/react-native: ^7.4.0 => 7.5.0 
    @react-native-async-storage/async-storage: ^1.17.11 => 1.17.11 
    @react-native-community/eslint-config: ^2.0.0 => 2.0.0 
    @react-native-community/netinfo: ^9.3.6 => 9.3.7 
    @react-native-community/push-notification-ios: ^1.10.1 => 1.10.1 (1.0.3)
    @react-native-firebase/app: ^17.3.2 => 17.3.2 
    @react-native-masked-view/masked-view: ^0.2.8 => 0.2.8 
    @react-native-picker/picker: ^2.4.8 => 2.4.8 
    @react-navigation/bottom-tabs: ^6.5.5 => 6.5.7 
    @react-navigation/native: ^6.0.14 => 6.1.6 
    @react-navigation/native-stack: ^6.9.2 => 6.9.12 
    @react-navigation/stack: ^6.3.13 => 6.3.16 
    @reduxjs/toolkit: ^1.9.3 => 1.9.3 
    @reduxjs/toolkit-query:  1.0.0 
    @reduxjs/toolkit-query-react:  1.0.0 
    @rneui/base: ^4.0.0-rc.7 => 4.0.0-rc.7 
    @rneui/themed: ^4.0.0-rc.7 => 4.0.0-rc.7 
    @tsconfig/react-native: ^2.0.2 => 2.0.3 
    @types/jest: ^26.0.23 => 26.0.24 
    @types/lodash: ^4.14.191 => 4.14.191 
    @types/react: ^18.0.21 => 18.0.28 
    @types/react-native: ^0.70.6 => 0.70.11 
    @types/react-native-vector-icons: ^6.4.13 => 6.4.13 
    @types/react-test-renderer: ^18.0.0 => 18.0.0 
    @types/uuid: ^9.0.0 => 9.0.1 
    @typescript-eslint/eslint-plugin: ^5.37.0 => 5.53.0 (3.10.1)
    @typescript-eslint/parser: ^5.37.0 => 5.53.0 (3.10.1)
    HelloWorld:  0.0.1 
    amazon-cognito-identity-js: ^5.2.12 => 5.2.14 (6.1.2)
    apollo3-cache-persist: ^0.14.1 => 0.14.1 
    aws-amplify: ^5.0.20 => 5.0.20 
    aws-amplify-react-native: ^7.0.2 => 7.0.2 
    babel-jest: ^26.6.3 => 26.6.3 
    babel-plugin-module-resolver: ^5.0.0 => 5.0.0 
    date-fns: ^2.29.3 => 2.29.3 
    eslint: ^7.32.0 => 7.32.0 
    eslint-config-prettier: ^8.6.0 => 8.6.0 (6.15.0)
    eslint-plugin-import: ^2.27.5 => 2.27.5 
    eslint-plugin-prettier: ^4.2.1 => 4.2.1 (3.1.2)
    eslint-plugin-react-hooks: ^4.6.0 => 4.6.0 
    example:  0.0.1 
    graphql: ^16.6.0 => 16.6.0 (15.8.0)
    hermes-inspector-msggen:  1.0.0 
    husky: ^8.0.0 => 8.0.3 
    jest: ^26.6.3 => 26.6.3 
    lint-staged: ^13.1.0 => 13.1.2 
    lodash: ^4.17.21 => 4.17.21 
    metro-react-native-babel-preset: 0.72.3 => 0.72.3 
    prettier: ^2.8.3 => 2.8.4 
    react: ^18.2.0 => 18.2.0 
    react-intl: ^6.2.8 => 6.2.10 
    react-native: 0.70.6 => 0.70.6 
    react-native-autoheight-webview: ^1.6.5 => 1.6.5 
    react-native-config: ^1.5.0 => 1.5.0 
    react-native-device-info: ^10.4.0 => 10.4.0 
    react-native-gesture-handler: ^2.9.0 => 2.9.0 
    react-native-get-random-values: ^1.8.0 => 1.8.0 
    react-native-gifted-chat: ^1.1.1 => 1.1.1 
    react-native-linear-gradient: ^2.6.2 => 2.6.2 
    react-native-safe-area-context: ^4.4.1 => 4.5.0 
    react-native-screens: ^3.18.2 => 3.20.0 
    react-native-secure-key-store: ^2.0.10 => 2.0.10 
    react-native-svg: ^13.8.0 => 13.8.0 
    react-native-svg-transformer: ^1.0.0 => 1.0.0 
    react-native-url-polyfill: ^1.3.0 => 1.3.0 
    react-native-vector-icons: ^9.2.0 => 9.2.0 
    react-native-webview: ^11.26.1 => 11.26.1 
    react-redux: ^8.0.5 => 8.0.5 
    react-test-renderer: 18.1.0 => 18.1.0 
    reactotron-react-native: ^5.0.3 => 5.0.3 
    typescript: ^4.8.3 => 4.9.5 
    uuid: ^9.0.0 => 9.0.0 (3.4.0, 8.3.2, 7.0.3)
  npmGlobalPackages:
    @aws-amplify/cli: 10.6.2
    corepack: 0.10.0
    npm: 8.5.1
    typescript: 4.6.4
    yarn: 1.22.19

```

</details>

### Describe the bug

1. I run my application in ios simulator
2. The error appears regardless either I give PN permission or not, but there is a random delay, sometimes I see this error immediately after I "accept" permission request, sometimes there is a delay

⚠️ You will see further in the code that I'm using `withAuthenticator`. So if I remove usage of it, error disappears.

```
Object {
  "EventsResponse": Object {
    "Results": Object {
      "357b9e00-c3ee-11ed-882b-ad5ebfd44bbb": Object {
        "EventsItemResponse": Object {
          "undefined": Object {
            "Message": "Event type cannot be null, empty or blank",
            "StatusCode": 400,
          },
        },
      },
    },
  },
}
```

P.S. Error appears while running both on simulator and real device

P.P.S. Despite the error I'm able to:
- get token in `onRegistry` call
- get notifications sent from AWS Pinpoint

### Expected behavior

Error isn't thrown

### Reproduction steps

1. See `package.json`:

  <details>
  
  ```
  {
    "name": "xxx",
    "version": "0.0.1",
    "private": true,
    "scripts": {
      ...
    },
    "lint-staged": {
      "*.{js,jsx,ts,tsx}": [
        "prettier --write",
        "eslint --fix"
      ]
    },
    "dependencies": {
      "@apollo/client": "^3.7.7",
      "@aws-amplify/pushnotification": "^5.0.20",
      "@aws-sdk/client-lex-models-v2": "^3.262.0",
      "@aws-sdk/client-lex-runtime-v2": "^3.262.0",
      "@notifee/react-native": "^7.4.0",
      "@react-native-async-storage/async-storage": "^1.17.11",
      "@react-native-community/netinfo": "^9.3.6",
      "@react-native-community/push-notification-ios": "^1.10.1",
      "@react-native-firebase/app": "^17.3.2",
      "@react-native-masked-view/masked-view": "^0.2.8",
      "@react-native-picker/picker": "^2.4.8",
      "@react-navigation/bottom-tabs": "^6.5.5",
      "@react-navigation/native": "^6.0.14",
      "@react-navigation/native-stack": "^6.9.2",
      "@react-navigation/stack": "^6.3.13",
      "@reduxjs/toolkit": "^1.9.3",
      "@rneui/base": "^4.0.0-rc.7",
      "@rneui/themed": "^4.0.0-rc.7",
      "amazon-cognito-identity-js": "^5.2.12",
      "apollo3-cache-persist": "^0.14.1",
      "aws-amplify": "^5.0.20",
      "aws-amplify-react-native": "^7.0.2",
      "date-fns": "^2.29.3",
      "graphql": "^16.6.0",
      "lodash": "^4.17.21",
      "react": "^18.2.0",
      "react-intl": "^6.2.8",
      "react-native": "0.70.6",
      "react-native-autoheight-webview": "^1.6.5",
      "react-native-config": "^1.5.0",
      "react-native-device-info": "^10.4.0",
      "react-native-gesture-handler": "^2.9.0",
      "react-native-get-random-values": "^1.8.0",
      "react-native-gifted-chat": "^1.1.1",
      "react-native-linear-gradient": "^2.6.2",
      "react-native-safe-area-context": "^4.4.1",
      "react-native-screens": "^3.18.2",
      "react-native-secure-key-store": "^2.0.10",
      "react-native-svg": "^13.8.0",
      "react-native-url-polyfill": "^1.3.0",
      "react-native-vector-icons": "^9.2.0",
      "react-native-webview": "^11.26.1",
      "react-redux": "^8.0.5",
      "uuid": "^9.0.0"
    },
    "devDependencies": {
      "@babel/core": "^7.12.9",
      "@babel/runtime": "^7.12.5",
      "@react-native-community/eslint-config": "^2.0.0",
      "@tsconfig/react-native": "^2.0.2",
      "@types/jest": "^26.0.23",
      "@types/lodash": "^4.14.191",
      "@types/react": "^18.0.21",
      "@types/react-native": "^0.70.6",
      "@types/react-native-vector-icons": "^6.4.13",
      "@types/react-test-renderer": "^18.0.0",
      "@types/uuid": "^9.0.0",
      "@typescript-eslint/eslint-plugin": "^5.37.0",
      "@typescript-eslint/parser": "^5.37.0",
      "babel-jest": "^26.6.3",
      "babel-plugin-module-resolver": "^5.0.0",
      "eslint": "^7.32.0",
      "eslint-config-prettier": "^8.6.0",
      "eslint-plugin-import": "^2.27.5",
      "eslint-plugin-prettier": "^4.2.1",
      "eslint-plugin-react-hooks": "^4.6.0",
      "husky": "^8.0.0",
      "jest": "^26.6.3",
      "lint-staged": "^13.1.0",
      "metro-react-native-babel-preset": "0.72.3",
      "prettier": "^2.8.3",
      "react-native-svg-transformer": "^1.0.0",
      "react-test-renderer": "18.1.0",
      "reactotron-react-native": "^5.0.3",
      "typescript": "^4.8.3"
    },
    "jest": {
      "preset": "react-native",
      "moduleFileExtensions": [
        "ts",
        "tsx",
        "js",
        "jsx",
        "json",
        "node"
      ]
    }
  }
  ```
  
  </details>

2. add push notifications with amplify 

3. `AppDelegate.h`

<details>

```
#import <UserNotifications/UNUserNotificationCenter.h>
#import <React/RCTBridgeDelegate.h>
#import <UIKit/UIKit.h>

@interface AppDelegate : UIResponder <UIApplicationDelegate, RCTBridgeDelegate, UNUserNotificationCenterDelegate>

@property(nonatomic, strong) UIWindow *window;

@end
```

</details>

4. `AppDelegate.mm`

<details>

```
#import <UserNotifications/UserNotifications.h>
#import <RNCPushNotificationIOS.h>

#import "AppDelegate.h"

#import <Firebase.h>

#import <React/RCTBridge.h>
#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

#import <React/RCTAppSetupUtils.h>

#if RCT_NEW_ARCH_ENABLED
#import <React/CoreModulesPlugins.h>
#import <React/RCTCxxBridgeDelegate.h>
#import <React/RCTFabricSurfaceHostingProxyRootView.h>
#import <React/RCTSurfacePresenter.h>
#import <React/RCTSurfacePresenterBridgeAdapter.h>
#import <ReactCommon/RCTTurboModuleManager.h>

#import <react/config/ReactNativeConfig.h>

static NSString *const kRNConcurrentRoot = @"concurrentRoot";

@interface AppDelegate () <RCTCxxBridgeDelegate, RCTTurboModuleManagerDelegate> {
  RCTTurboModuleManager *_turboModuleManager;
  RCTSurfacePresenterBridgeAdapter *_bridgeAdapter;
  std::shared_ptr<const facebook::react::ReactNativeConfig> _reactNativeConfig;
  facebook::react::ContextContainer::Shared _contextContainer;
}
@end
#endif

@implementation AppDelegate

// Required for the register event.
- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
{
 [RNCPushNotificationIOS didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
}
// Required for the notification event. You must call the completion handler after handling the remote notification.
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo
fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler
{
  [RNCPushNotificationIOS didReceiveRemoteNotification:userInfo fetchCompletionHandler:completionHandler];
}
// Required for the registrationError event.
- (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error
{
 [RNCPushNotificationIOS didFailToRegisterForRemoteNotificationsWithError:error];
}
// Required for localNotification event
- (void)userNotificationCenter:(UNUserNotificationCenter *)center
didReceiveNotificationResponse:(UNNotificationResponse *)response
         withCompletionHandler:(void (^)(void))completionHandler
{
  [RNCPushNotificationIOS didReceiveNotificationResponse:response];
}

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  [FIRApp configure];
  
  RCTAppSetupPrepareApp(application);

  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];

#if RCT_NEW_ARCH_ENABLED
  _contextContainer = std::make_shared<facebook::react::ContextContainer const>();
  _reactNativeConfig = std::make_shared<facebook::react::EmptyReactNativeConfig const>();
  _contextContainer->insert("ReactNativeConfig", _reactNativeConfig);
  _bridgeAdapter = [[RCTSurfacePresenterBridgeAdapter alloc] initWithBridge:bridge contextContainer:_contextContainer];
  bridge.surfacePresenter = _bridgeAdapter.surfacePresenter;
#endif

  NSDictionary *initProps = [self prepareInitialProps];
  UIView *rootView = RCTAppSetupDefaultRootView(bridge, @"AstuvetMobile", initProps);

  if (@available(iOS 13.0, *)) {
    rootView.backgroundColor = [UIColor systemBackgroundColor];
  } else {
    rootView.backgroundColor = [UIColor whiteColor];
  }

  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

   // Define UNUserNotificationCenter
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;

  return YES;
}

//Called when a notification is delivered to a foreground app.
-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
{
  completionHandler(UNNotificationPresentationOptionSound | UNNotificationPresentationOptionAlert | UNNotificationPresentationOptionBadge);
}

/// This method controls whether the `concurrentRoot`feature of React18 is turned on or off.
///
/// @see: https://reactjs.org/blog/2022/03/29/react-v18.html
/// @note: This requires to be rendering on Fabric (i.e. on the New Architecture).
/// @return: `true` if the `concurrentRoot` feture is enabled. Otherwise, it returns `false`.
- (BOOL)concurrentRootEnabled
{
  // Switch this bool to turn on and off the concurrent root
  return true;
}

- (NSDictionary *)prepareInitialProps
{
  NSMutableDictionary *initProps = [NSMutableDictionary new];

#ifdef RCT_NEW_ARCH_ENABLED
  initProps[kRNConcurrentRoot] = @([self concurrentRootEnabled]);
#endif

  return initProps;
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

#if RCT_NEW_ARCH_ENABLED

#pragma mark - RCTCxxBridgeDelegate

- (std::unique_ptr<facebook::react::JSExecutorFactory>)jsExecutorFactoryForBridge:(RCTBridge *)bridge
{
  _turboModuleManager = [[RCTTurboModuleManager alloc] initWithBridge:bridge
                                                             delegate:self
                                                            jsInvoker:bridge.jsCallInvoker];
  return RCTAppSetupDefaultJsExecutorFactory(bridge, _turboModuleManager);
}

#pragma mark RCTTurboModuleManagerDelegate

- (Class)getModuleClassFromName:(const char *)name
{
  return RCTCoreModulesClassProvider(name);
}

- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
                                                      jsInvoker:(std::shared_ptr<facebook::react::CallInvoker>)jsInvoker
{
  return nullptr;
}

- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
                                                     initParams:
                                                         (const facebook::react::ObjCTurboModule::InitParams &)params
{
  return nullptr;
}

- (id<RCTTurboModule>)getModuleInstanceFromClass:(Class)moduleClass
{
  return RCTAppSetupDefaultModuleFromClass(moduleClass);
}

#endif

@end
```

</details>

5. `ios/Podfile` (part of)

<details>

```
use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => false, # false is set for debug purpose: https://github.com/facebook/react-native/issues/34624
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => FlipperConfiguration.enabled,
    # :flipper_configuration => FlipperConfiguration.enabled, # disabled because we need use_frameworks for firebase
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  use_frameworks! :linkage => :static
  $RNFirebaseAsStaticFramework = true

  target 'AstuvetMobileTests' do
    inherit! :complete
    # Pods for testing
  end
  post_install do |installer|
    react_native_post_install(
      installer,
      # Set `mac_catalyst_enabled` to `true` in order to apply patches
      # necessary for Mac Catalyst builds
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end
```

</details>

6. `index.js` (root file) looks like the following:

<details>

```
import { AppRegistry } from 'react-native';
import { Amplify } from 'aws-amplify';
import { PushNotification } from '@aws-amplify/pushnotification';
import App from './src/modules/App/App.tsx';
import { name as appName } from './app.json';

import awsconfig from './src/aws-exports';

Amplify.configure(awsconfig);

PushNotification.onRegister(async token => {
  await AsyncStorage.setItem('TOKEN', token);
  console.log(`onRegistry token: ${token}`);
});

PushNotification.onNotification(notification => {
  if (notification.foreground) {
    console.log(`onNotification foreground: ${notification}`);
  } else {
    console.log(`onNotification background or closed: ${notification}`);
  }
});

PushNotification.onNotificationOpened(notification => {
  console.log(`onNotificationOpened: ${notification} `);
});

AppRegistry.registerComponent(appName, () => App);

```

</details>

7. `App` looks this way:

<details>

```
import 'react-native-gesture-handler';
import 'react-native-url-polyfill/auto';
import 'react-native-get-random-values';

import { ThemeProvider } from '@rneui/themed';
import { IWithAuthenticatorProps } from 'aws-amplify-react-native';
import { useState } from 'react';
import { View } from 'react-native';
import { Provider } from 'react-redux';

import { Compose } from '~/components/Compose';
import { ActivityIndicator } from '~/components/ui';
import { useAsyncEffect } from '~/hooks';
import { AdminClientProvider } from '~/lib/adminClient';
import { store } from '~/lib/store';
import { theme } from '~/lib/theme';

import { TranslateProvider } from '../../lib/i18n';
import { useLoadUser } from '../Auth';
import { UserProvider } from '../Auth/UserProvider';
import { withAuthenticator } from '../Auth/withAuthenticator';

import { NavigationProvider } from './Navigation/NavigationProvider';

const AppWithAuthenticator = withAuthenticator(() => {
  return <View />;
}) as unknown as React.ComponentClass<IWithAuthenticatorProps>;

const App = () => {
  return (
    <Provider store={store}>
      <Compose components={[TranslateProvider, NavigationProvider]}>
        <AdminClientProvider>
          <ThemeProvider theme={theme}>
            <AppWithAuthenticator />
          </ThemeProvider>
        </AdminClientProvider>
      </Compose>
    </Provider>
  );
};

export default App;
```

</details>

8. `withAuthenticator`

<details>

```
import { withAuthenticator as withBaseAuthenticator } from 'aws-amplify-react-native';

import { authTheme } from '~/lib/theme';

const signUpConfig = {
  header: 'Sign Up',
  hideAllDefaults: true,
  defaultCountryCode: '1',
  signUpFields: [
    {
      label: 'E-mail',
      key: 'email',
      required: true,
      displayOrder: 1,
      type: 'string',
    },
    {
      label: 'Password',
      key: 'password',
      required: true,
      displayOrder: 2,
      type: 'password',
    },
    {
      label: 'Name',
      key: 'name',
      required: true,
      displayOrder: 3,
      type: 'string',
    },
    {
      label: 'Family Name',
      key: 'family_name',
      required: true,
      displayOrder: 4,
      type: 'string',
    },
    {
      label: 'Phone Number',
      key: 'phone_number',
      required: false,
      displayOrder: 5,
      type: 'string',
    },
  ],
};

export const withAuthenticator = (App: React.ComponentType<object>) => {
  return withBaseAuthenticator(App, {
    theme: authTheme,
    signUpConfig,
    usernameAttributes: 'E-mail',
  });
};

```

</details>

**What I've noticed**

Bug appears only if `withAuthenticator` is used. If `App` component is a blank `View` - nothing happens.

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

const awsmobile = {
    "aws_project_region": "us-east-1",
    "aws_appsync_graphqlEndpoint": "https://5o6pgczqafetfoncwcuutow43u.appsync-api.us-east-1.amazonaws.com/graphql",
    "aws_appsync_region": "us-east-1",
    "aws_appsync_authenticationType": "API_KEY",
    "aws_appsync_apiKey": "da2-xxxx", 
    "aws_cognito_identity_pool_id": "us-east-1:d8c41199-6839-4e4c-91ee-5495b5f4402e",
    "aws_cognito_region": "us-east-1",
    "aws_user_pools_id": "us-east-1_Pfa4jHXjO",
    "aws_user_pools_web_client_id": "2j1up24l0akrhrsdpg53r1as0c",
    "oauth": {},
    "aws_cognito_username_attributes": [
        "EMAIL"
    ],
    "aws_cognito_social_providers": [],
    "aws_cognito_signup_attributes": [
        "NAME",
        "FAMILY_NAME"
    ],
    "aws_cognito_mfa_configuration": "OFF",
    "aws_cognito_mfa_types": [
        "SMS"
    ],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": []
    },
    "aws_cognito_verification_mechanisms": [
        "EMAIL"
    ],
    "aws_mobile_analytics_app_id": "d469fb29e45742a7b4fe1779ee798140",
    "aws_mobile_analytics_app_region": "us-east-1"
};

### Additional configuration

```json
{
    "UserPool": {
        "Id": "us-east-1_xxx",
        "Name": "AstuvetMobile-dev",
        "Policies": {
            "PasswordPolicy": {
                "MinimumLength": 8,
                "RequireUppercase": false,
                "RequireLowercase": false,
                "RequireNumbers": false,
                "RequireSymbols": false,
                "TemporaryPasswordValidityDays": 7
            }
        },
        "LambdaConfig": {},
        "LastModifiedDate": "2022-12-28T14:55:40.379000+04:00",
        "CreationDate": "2022-12-28T14:55:40.379000+04:00",
        "SchemaAttributes": [
            {
                "Name": "sub",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": false,
                "Required": true,
                "StringAttributeConstraints": {
                    "MinLength": "1",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "name",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": true,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "given_name",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "family_name",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": true,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "middle_name",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "nickname",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "preferred_username",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "profile",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "picture",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "website",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "email",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "email_verified",
                "AttributeDataType": "Boolean",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false
            },
            {
                "Name": "gender",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "birthdate",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "10",
                    "MaxLength": "10"
                }
            },
            {
                "Name": "zoneinfo",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "locale",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "phone_number",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "phone_number_verified",
                "AttributeDataType": "Boolean",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false
            },
            {
                "Name": "address",
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "StringAttributeConstraints": {
                    "MinLength": "0",
                    "MaxLength": "2048"
                }
            },
            {
                "Name": "updated_at",
                "AttributeDataType": "Number",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Required": false,
                "NumberAttributeConstraints": {
                    "MinValue": "0"
                }
            }
        ],
        "AutoVerifiedAttributes": [
            "email"
        ],
        "UsernameAttributes": [
            "email"
        ],
        "EmailVerificationMessage": "Your verification code is {####}",
        "EmailVerificationSubject": "Your verification code",
        "VerificationMessageTemplate": {
            "EmailMessage": "Your verification code is {####}",
            "EmailSubject": "Your verification code",
            "DefaultEmailOption": "CONFIRM_WITH_CODE"
        },
        "MfaConfiguration": "OFF",
        "EstimatedNumberOfUsers": 20,
        "EmailConfiguration": {
            "EmailSendingAccount": "COGNITO_DEFAULT"
        },
        "UserPoolTags": {},
        "AdminCreateUserConfig": {
            "AllowAdminCreateUserOnly": false,
            "UnusedAccountValidityDays": 7
        },
        "UsernameConfiguration": {
            "CaseSensitive": false
        },
        "Arn": "arn:aws:cognito-idp:us-east-xxx"
    }
}
```

### Mobile Device

iPhone 14

### Mobile Operating System

iOS 16.3.1

**Comment by tannerabread** [comment]
Hi @tyndria I tried reproducing your issue a couple of different ways with no luck yet, I did notice you are using the old UI package for React Native where we have a [new React Native UI package](https://ui.docs.amplify.aws/react-native/connected-components/authenticator) that is in dev preview that might work better here

I'll keep trying to reproduce this if that doesn't fix your problem but I did not see that error on a barebones app using `withAuthenticator` how you were, and without seeing your entire app I can't really add more features in. I do appreciate the details steps you have given so far

**Comment by tyndria** [comment]
@tannerabread it seems you were right, no error with new UI package

I will keep posted here If I ever see this message again

thanks!

**Comment by tannerabread** [comment]
Glad that unblocked you! I'll close this for now but just comment here or create a new issue if it does come back

Thanks!

**Comment by tannerabread** [comment]
Just an update, but while trying to check on the other issue you opened #11110 and make sure the solution I mentioned works, I did forget to switch to the new UI package and got this same error on an android emulator.

**Comment by tyndria** [comment]
Thanks for update! It's working with new package indeed! And I glad you were able to reproduce it
